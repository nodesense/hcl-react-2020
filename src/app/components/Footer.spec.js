import React from 'react';
import Footer from './Footer';
import renderer from 'react-test-renderer';

import {BrowserRouter as Router} from 'react-router-dom';

// test is like test suite + 1 test case inside
test('Link changes the class when hovered', () => {
  const component = renderer.create(
      <Router>
        <Footer companyName="HCL" 
            year={2020}
            branches = { ['IN', 'UK'] }
            > </Footer>
    </Router>        
            ,
  );
  // convert Virtual DOM into JSON structure
  let tree = component.toJSON();
  // create a file in the project very first time
  // second time onwards, uses snapshot file against test result
  expect(tree).toMatchSnapshot();
})