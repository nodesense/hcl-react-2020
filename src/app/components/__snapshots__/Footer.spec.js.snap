// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div>
  <hr />
  <p>
    Copyrights @
    HCL
    , 
    2020
  </p>
  <ul>
    <li>
      IN
    </li>
    <li>
      UK
    </li>
  </ul>
   
  <button
    className="success"
  >
    தொடர்பு
  </button>
  <button
    onClick={[Function]}
  >
    Cart
  </button>
</div>
`;
